### PR TITLE
#1016 IconCell GenericChooseModal components - Duplicate #1063

### LINKED ISSUE #1016: IconCell component and generic choose modal updates from vaccines branch
## Describe

In the `vaccine-development` branch there is an `IconCell` component and updates to the component `GenericChooseModal` which would be helpful to have in development/master branches

### Version and device info
 - App version: 2.3.3

### Additional context
Vaccine branch is not going to be ready to merge into development for a little while

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #1016 

## Change summary
Duplicate of #1063 based off master

### CODE REVIEW

File: src/widgets/SequentialSteps.js [hunk 1]
```diff
@@ -144,6 +144,8 @@ SequentialSteps.propTypes = {
   currentStepKey: PropTypes.string.isRequired,
 };
 
+export default SequentialSteps;
```
Comment (Chris-Petty) [inline src/widgets/SequentialSteps.js]: Redundant?

File: src/widgets/modals/StocktakeBatchModal.js [hunk 2]
```diff
@@ -226,21 +239,38 @@ export default class StocktakeBatchModal extends React.Component {
...
-        title={modalStrings.select_a_reason}
-      />
-    );
+  renderModalContent = () => {
+    const { modalKey, currentBatch, reasons } = this.state;
+    const { option } = currentBatch;
+    const highlightValue = option && option.title;
+    const { REASON_EDIT } = MODAL_KEYS;
+    switch (modalKey) {
+      default: {
```
Comment (Chris-Petty) [inline src/widgets/modals/StocktakeBatchModal.js]: Stylistic choice, but I'd have put the default/null return as the last clause in the switch.

**Comment by Chris-Petty** [comment]
Works fine in testing 👍

File: src/widgets/modals/StocktakeBatchModal.js (hunk 2 above)
Comment (joshxg) [reply src/widgets/modals/StocktakeBatchModal.js]: c8198114c3267b88faad8799ce442c63694058cf

File: src/widgets/SequentialSteps.js (hunk 1 above)
Comment (joshxg) [reply src/widgets/SequentialSteps.js]: Yeah, all components seem to have this set up so I've just kept going with  for consistency..

**Pull request merged**